feature-gate/program/src/instruction.rs

```diff
@@ -0,0 +1,126 @@
+//! Program instructions
+
+use solana_program::{
+    feature::Feature,
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    rent::Rent,
+    system_instruction, system_program,
+};
+
+/// Feature Gate program instructions
+#[derive(Clone, Debug, PartialEq)]
+pub enum FeatureGateInstruction {
+    /// Submit a feature for activation.
+    ///
+    /// Note: This instruction expects the account to exist and be owned by the
+    /// system program. The account should also have enough rent-exempt lamports
+    /// to cover the cost of the account creation for a
+    /// `solana_program::feature::Feature` state prior to invoking this
+    /// instruction.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w+s]` Feature account (must be a system account)
+    ///   1. `[]` System program
+    Activate,
+    /// Revoke a pending feature activation.
```

Comment:
nit: You may want to specify what "pending" means, ie. a feature whose account has been allocated + assigned, but hasn't been processed by the network yet.